### DISCUSSION TITLE
Les redirections `client-side` ne fonctionnent pas

### DISCUSSION THREAD

**Discussion by maxgfr** [SocialGouv/code-du-travail-numerique#5530] [Technique]
## Problème

Les redirections côté clients avec les `router.push` et `Link` ne fonctionnent pas 

Lien de la documentation de next.js : https://nextjs.org/docs/app/api-reference/next-config-js/redirects

![image](https://github.com/SocialGouv/code-du-travail-numerique/assets/25312957/ae0ce31c-4970-4bfa-ae98-b68309789112)

## Contexte

Le fichier `redirect.json` fonctionne bien sur les redirections côtés serveur. (ex: arrivée de puis google)

## Solutions

La difficulté provient de l'interpréteur de regex (ex: `3248-:slug`) qui permet de manipuler au sein d'un json les rédirections nécessaires.

J'ai essayé de voir s'il était possible de récupérer la fonction qu'ils utilisent de leur côté pour parser les string afin qu'on puisse les ré-utiliser ça de notre côté. A ma connaissance, non ce n'est pas possible.

L'autre solution, ça serait de reprendre ce qu'on utilisait auparavant : https://github.com/SocialGouv/code-du-travail-numerique/pull/4886

**Comment by carolineBda** [comment]
Sinon on ajoute à la racine un middleware qui utilise le même fichier redirect.json ?
https://nextjs.org/docs/app/building-your-application/routing/middleware#matcher
